### PR TITLE
add cpp euler tour implementation to usaco 696

### PR DESCRIPTION
**Pull request by WilliamYuan42**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [X] I have tested my code.
- [X] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [X] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: solutions/gold/usaco-696.mdx [hunk 1]
```diff
@@ -99,8 +99,81 @@ public class PromotionCounting {
...
+	while (i <= n) {
+		bit[i] += v;
+		i += i & -i;
+	}
+}
+
+void dfs(int cur, int prev) {
+	in[cur] = t;
+	for (int nxt : adj[cur])
+		if (nxt != prev) dfs(nxt, cur);
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-696.mdx:133]: braces

File: solutions/gold/usaco-696.mdx [hunk 2]
```diff
@@ -99,8 +99,81 @@ public class PromotionCounting {
...
 </JavaSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+const int mx = 1e5 + 1;
+int n, t = 0, out[mx], in[mx], bit[mx], ans[mx];
+vector<int> adj[mx];
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-696.mdx:110]: just...use vectors...

File: solutions/gold/usaco-696.mdx [hunk 3]
```diff
@@ -99,8 +99,81 @@ public class PromotionCounting {
...
+	return res;
+}
+
+void update(int i, int v) {
+	i++;
+	while (i <= n) {
+		bit[i] += v;
+		i += i & -i;
+	}
+}
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-696.mdx:128]: use a struct for bits
[this](https://usaco.guide/gold/PURS?lang=cpp#solution---dynamic-range-sum-queries-with-a-bit) part has a good impl

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/gold/usaco-696.mdx (hunk 2 above)
Comment (WilliamYuan42) [reply solutions/gold/usaco-696.mdx]: only using vectors for bit and ans for convenience; i'd rather not have to add out, in, adj as arguments for dfs

Comment (SansPapyrus683) [reply solutions/gold/usaco-696.mdx]: what's wrong with declaring vectors globally

File: solutions/gold/usaco-696.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/gold/usaco-696.mdx:138]: not fixed

File: solutions/gold/usaco-696.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/gold/usaco-696.mdx]: also uh put the declarations on separate lines

File: solutions/gold/usaco-696.mdx (hunk 1 above)
Comment (WilliamYuan42) [reply solutions/gold/usaco-696.mdx:138]: i could've sworn i added braces 💀

File: solutions/gold/usaco-696.mdx (hunk 2 above)
Comment (WilliamYuan42) [reply solutions/gold/usaco-696.mdx]: either i'd have to resize it within main() (more imo unnecessary code) or i set it with size mx in the global space anyway(pointless?)

Comment (WilliamYuan42) [reply solutions/gold/usaco-696.mdx]: nvm

File: solutions/gold/usaco-696.mdx [hunk 4]
```diff
@@ -98,16 +99,101 @@ public class PromotionCounting {
...
 ```
+
 </JavaSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int mx = 1e5 + 1;
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-696.mdx:110]: is the +1 necessary

File: solutions/gold/usaco-696.mdx [hunk 5]
```diff
@@ -98,16 +99,102 @@ public class PromotionCounting {
...
 </JavaSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int mx = 1e5;
+int n, t = 0;
+vector<int> adj[mx], out(mx), in(mx);
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-696.mdx:112]: rename to MAX_N

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
did you test this code?

File: solutions/gold/usaco-696.mdx (hunk 5 above)
Comment (WilliamYuan42) [reply solutions/gold/usaco-696.mdx:112]: yeah i tested the code

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/gold/usaco-696.mdx [hunk 6]
```diff
@@ -98,16 +99,102 @@ public class PromotionCounting {
...
+			    bit.sum(out[a[r].second]) - bit.sum(in[a[r].second] - 1);
+			r++;
+		}
+
+		for (int j = i; j < r; j++) bit.update(out[a[j].second], 1);
+
+		i = r;
+	}
+
+	freopen("promote.out", "w", stdout);
```
Comment (ryanchou-dev) [inline solutions/gold/usaco-696.mdx:183]: just put this next to the other freopen

**Review by ryanchou-dev — APPROVED** [review]

File: solutions/gold/usaco-696.mdx (hunk 6 above)
Comment (SansPapyrus683) [reply solutions/gold/usaco-696.mdx:183]: that was me lol

**Pull request merged**